cpp/src/parquet/stream_reader_test.cc

```diff
@@ -465,5 +531,151 @@ TEST_F(TestStreamReader, SkipColumns) {
   ASSERT_EQ(0, reader_.SkipColumns(100));
 }
 
+class TestOptionalFields : public ::testing::Test {
```

Comment:
I think it would have been easier to read and understand each test if there was one per type, and then final one that tests multiple column conditions.

Reply:
I think it is better to have a single schema with all the types that need to be tested.   Otherwise, this will require a lot of setup code for each of the different schemas.

Reply:
Couldn't that be done with a helper method?

Reply:
I agree that the code has some feeling of boilerplate / DRY issues. For example tests could be written with TEST_P and parameterized on data type.

For example: https://www.diffchecker.com/NaI5iMe9.